lib/ledger_sync/resource.rb

```diff
@@ -23,8 +23,14 @@ def initialize(external_id: nil, ledger_id: nil, sync_token: nil, **data)
       @sync_token = sync_token
 
       data.each do |attr_key, val|
-        if (self.class.references || {}).key?(attr_key)
-          raise "#{val} must be of type #{self.class.references[attr_key]}" if self.class.references[attr_key] != val.class
+        if (self.class.references_one || {}).key?(attr_key)
```

Comment:
It may not hurt us to just adopt standard `belongs_to` and `has_many` as they do operate the same as Rails, right?